subt/ros/robot/src/filter_pointcloud.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python
+"""
+  Filter PointCloud2 received from X4 drone for Octomap server
+  - remove nearby propellers (up to 31cm) in the RGBD camera view
+  - replace xyz = (inf, inf, inf) to range larger than 10m
+"""
+# http://docs.ros.org/en/melodic/api/sensor_msgs/html/msg/PointCloud2.html
+# http://docs.ros.org/en/api/sensor_msgs/html/point__cloud2_8py_source.html
+import ctypes
+import struct
+
+import rospy
+from sensor_msgs.msg import PointCloud2, PointField
+
+import numpy as np
+
+
+class FilterPointCloud:
+    def __init__(self):
+#        self.points_subscriber = rospy.Subscriber("/input_points", PointCloud2, self.points_callback)
+        self.points_subscriber = rospy.Subscriber("/A100LXM/rgbd_camera/depth/points", PointCloud2, self.points_callback)
+#        self.points_publisher = rospy.Publisher("/output_points", PointCloud2, queue_size = 1)
+        self.points_publisher = rospy.Publisher("/points_cleaned", PointCloud2, queue_size=1)
+
+        # create flat "infinity" background
+        size = 640 * 480
+        self.inf_x = np.zeros(size, dtype=np.float32)
+        self.inf_y = np.zeros(size, dtype=np.float32)
+        self.inf_z = np.zeros(size, dtype=np.float32)
+        fx = 554.25469
+        MAX_VALUE = 11.0
+        for x in range(640):
+            for y in range(480):
```

Comment:
There must be a better way.

DepthParams in depth.py initialize a similar vector. If you take that code, multiply by MAX_VALUE at the end, you should get what you are looking for, no?

Reply:
I am sure there is other numpy way - this is initialized just once. There were so many "details" wrong that I preferred stupid version ... again for me it is hard to check that this particular code works 1:1